astropy/units/quantity_helper/function_helpers.py

```diff
@@ -252,15 +251,63 @@ def like_helper(a, *args, **kwargs):
     return (a.view(np.ndarray),) + args, kwargs, unit, None
 
 
+def _nanop_helper(a, *, out, _out_unit_from_a):
```

Comment:
Hmm, this is not how `ufuncs` treat `out`, so will need a bit of adjustment. Basically, if `out` is given, the unit should just be overwritten (just like the data are), so `out_unit` is just the unit of `a`, or dimensionless if `a` is not a `Quantity` (which is possible; numpy functions dispatch on both input and output arrays).

Reply:
> numpy functions dispatch on both input and output arrays

this is actually why I needed to factor in `out`'s units: if not, calling `np.nanvar(q, out=o)` (where `q` and `o` are `Quantity`s) would first be processed `nanvar`'s helper (which strips out `q`'s unit) and *then* through `nanstd`'s helper (because `o` is still a quantity), but then `o` is the only place where the output unit is still available.